src/node-jspm-jasmine.js

```diff
@@ -88,26 +85,67 @@ export function runTests(opts, errCallback = function() {}) {
-							load.source  + '\n' + inlineSourceMap(load.metadata.sourceMap)
+			// create systemjs hook to allow Istanbul to instrument transpiled sources
+			const instrument = new Instrumenter();
+			const systemInstantiate = SystemJS.instantiate;
+			const tempDirectory = path.join(__dirname, '../tmp/');
+			// "instantiate" is the hook that provides the transpiled source
+			SystemJS.instantiate = function(load) {
+				try {
+					// create a unique key to store the sources of modules for the browser
+					const fileKey = getFileKey(getOSFilePath(load.address), process.cwd());
+					// exclude the dependency modules (i.e. libraries) from instrumentation
+					if (coverageFiles[fileKey]) {
+						let filename
+						// arrange sourcemaps
+						if (load.metadata.sourceMap) {
+							// put file's transpiled counterpart in temp folder
+        					filename = path.join(tempDirectory, encodeURIComponent(fileKey));
+							// keeping sourcesContent causes duplicate reports
+							delete load.metadata.sourceMap.sourcesContent
+							// this is the file being "instrumented"
+							load.metadata.sourceMap.file = tempDirectory + encodeURIComponent(fileKey)
+							// removing "file://" from paths
+							load.metadata.sourceMap.sources = load.metadata.sourceMap.sources.map(
+								filename => getOSFilePath(filename)
+							)
+							// write transpiled file with an inlined-sourceMap to temp directory
+							fs.writeFileSync(
+								filename,
+								load.source  + '\n' + inlineSourceMap(load.metadata.sourceMap)
+							)
+						}  else {
+							// keep file as is since it doesn't have a sourcemap to be remaped
+							filename = getOSFilePath(load.address)
+						}
+						// instrument with istanbul
+						load.source = instrument.instrumentSync(
+							load.source,
+							// make the path-like file key into something that can be used as a name
+							filename
 						)
-						normalizedAddress += tranpiledFileSuffix;
+						// files that are not included as dependencies in the spec
+						// still need to be evaluated. So here, we're going to exclude
+						// files that are already being imported by the specs, leaving
+						// only files that were not imported present in the "coverageFiles",
+						// which will then be imported (evaluated) when the tests conclude.
+						delete coverageFiles[fileKey]
 					}
-					load.source = Instrument.instrumentSync(load.source, normalizedAddress)
+				} catch (ex) {
+					errCallback(ex)
 				}
-				return systemInstantiate.call(SystemJS, load);
+				// call the original "instantiate" hook function
+				return systemInstantiate.call(SystemJS, load)
+			}
+
+			// storing related variables in the "opts.coverage" object to avoid
+			// having to pass more arguments to the "importTestFiles" function
+			opts.coverage.coverageFiles = coverageFiles
+			opts.coverage.tempDirectory = tempDirectory
+
+			// create temp directory
+			if (!fs.existsSync(tempDirectory)) {
```

Comment:
@joeldenning All things considered, it shouldn't really matter because all the files contained within will be overwritten, followed by a whole folder deletion if all goes well. But yeah, I'll update it to clean it up every time, doesn't hurt to.